In [14]:
import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, r2_score, mean_squared_error
import numpy as np

In [15]:
df = pd.read_csv('../data/processed/elevator_clean.csv')

In [16]:
df.head()

,Unnamed: 0,ID,revolutions,humidity,vibration,x1,x2,x3,x4,x5,Hour_Bin
0,0,1,93.744,73.999,18.0,167.743,19.745,1.266828,8787.937536,5475.852001,1
1,1,2,93.740,73.999,18.0,167.739,19.741,1.266774,8787.187600,5475.852001,1
2,2,3,93.736,73.998,18.0,167.734,19.738,1.266737,8786.437696,5475.704004,1
3,3,4,93.732,73.998,18.0,167.730,19.734,1.266683,8785.687824,5475.704004,1
4,4,5,93.729,73.998,18.0,167.727,19.731,1.266642,8785.125441,5475.704004,1


In [17]:
df.drop('Hour_Bin', axis=1, inplace=True)

In [18]:
df.head()

,Unnamed: 0,ID,revolutions,humidity,vibration,x1,x2,x3,x4,x5
0,0,1,93.744,73.999,18.0,167.743,19.745,1.266828,8787.937536,5475.852001
1,1,2,93.740,73.999,18.0,167.739,19.741,1.266774,8787.187600,5475.852001
2,2,3,93.736,73.998,18.0,167.734,19.738,1.266737,8786.437696,5475.704004
3,3,4,93.732,73.998,18.0,167.730,19.734,1.266683,8785.687824,5475.704004
4,4,5,93.729,73.998,18.0,167.727,19.731,1.266642,8785.125441,5475.704004


In [19]:
# Features and target
features = ["revolutions", "humidity", "x1", "x2", "x3", "x4", "x5"]
X = df[features]
y = df["vibration"]

In [20]:
split = int(0.80 * len(df))
X_train, X_test = X.iloc[:split], X.iloc[split:]
y_train, y_test = y.iloc[:split], y.iloc[split:]
 

In [21]:
model = LinearRegression()
model.fit(X_train, y_train)    # ← this is where learning happens

print("  Training complete!")
print()
print("  What the model learned (coefficients):")
for feature, coef in zip(features, model.coef_):
    print(f"    {feature:<14}: {coef:+.4f}")
print(f"    {'intercept':<14}: {model.intercept_:+.4f}")
 

  Training complete!

  What the model learned (coefficients):
    revolutions   : -0.2364
    humidity      : -188.9081
    x1            : -189.1446
    x2            : +188.6717
    x3            : +68.1474
    x4            : -0.0051
    x5            : +3.8457
    intercept     : +20909.1930


In [22]:
y_pred = model.predict(X_test)

In [23]:
# now we will calculate the mean absolute error and R^2 score to evaluate the model's performance
mae = mean_absolute_error(y_test, y_pred)
rmse = np.sqrt(mean_squared_error(y_test, y_pred))
r2 = r2_score(y_test, y_pred)

print(f"  Mean Absolute Error: {mae:.4f}")
print(f"  Root Mean Squared Error: {rmse:.4f}")
print(f"  R^2 Score: {r2:.4f}")

  Mean Absolute Error: 10.9689
  Root Mean Squared Error: 16.2670
  R^2 Score: -0.2390


####
As a baseline, we applied Linear Regression using all 7
  features. Under temporal evaluation, this model achieved
  MAE : 10.97, RMSE : 16.27 and R^2 score : -0.24
  The negative R² indicates the model performs worse than
  simply predicting the mean vibration for every row, confirming
  that the linear relationship between features and vibration
  is insufficient to capture the true dynamics."
####